src/signer.cpp

```diff
@@ -165,6 +176,38 @@ namespace sdk {
         return ec_sig_from_bytes(gsl::make_span(derived->priv_key).subspan(1), hash);
     }
 
+    priv_key_t software_signer::get_blinding_key_from_script(byte_span_t script)
+    {
+        GDK_RUNTIME_ASSERT(m_master_blinding_key.has_value());
+        const auto blinding_key = hmac_sha256(gsl::make_span(m_master_blinding_key.get()), script);
+        GDK_RUNTIME_ASSERT(ec_private_key_verify(blinding_key));
+        return blinding_key;
+    }
+
+    priv_key_t software_signer::derive_master_blinding_key(byte_span_t seed)
+    {
+        GDK_RUNTIME_ASSERT(m_net_params.liquid());
+
+        const std::string domain_str{ "Symmetric key seed" };
+        const auto domain = ustring_span(domain_str);
+        auto root = hmac_sha512(domain, seed);
+
+        const std::string label_str{ "SLIP-0077" };
+        const auto label = ustring_span(label_str);
+        auto node = hmac_sha512(gsl::make_span(root).first(EC_PRIVATE_KEY_LEN), label);
```

Comment:
`label` should be prefixed with the `'\x00'` byte before applying HMAC:
https://github.com/satoshilabs/slips/blob/master/slip-0021.md#child-node-derivation
https://github.com/satoshilabs/slips/pull/662/files#diff-a3d7c0a374f7cf006bee833349d8d689R42